src/data_structure/ziplist/ziplist.h

```diff
@@ -246,6 +246,17 @@ ziplist_rstatus_e ziplist_remove_val(uint32_t *removed, ziplist_p zl, const stru
  * CALLER MUST MAKE SURE THERE IS ENOUGH MEMORY!!!
  */
 ziplist_rstatus_e ziplist_insert(ziplist_p zl, struct blob *val, int64_t idx);
+/*
+ * trim list down to contain at most count entries, starting at idx. a negative count means
+ * starting from the end, and gives a list that is non inclusive of idx.
+ *
+ * e.g.
+ * if list contains { 0, 1, 2, 3, 4, 5 }, and we call ziplist_trim(zl, 4, -3),
+ * the trimmed list will be { 1, 2, 3 }.
+ *
+ * if there are fewer than count entries, all entries starting at idx are preserved
+ */
+ziplist_rstatus_e ziplist_trim(ziplist_p zl, int64_t idx, int64_t count);
```

Comment:
What do you think the relationship between `ziplist_remove` and `ziplist_trim` should be? I admit I haven't thought about this very carefully, but do you think trim will either enable or simplify implementation of commands that remove doesn't support well?

Reply:
I probably haven't thought about this carefully enough either. The most obvious benefit of `ziplist_trim` as currently implemented right now is it saves on one remove call if we are trimming in the middle of the list. But for other cases, it is essentially the same as remove. Having this API obviously simplifies trim commands, but I can't think of too many other uses for it at the moment.